lib/fastlane/actions/setup_jenkins.rb

```diff
@@ -0,0 +1,180 @@
+        if params[:output_directory]
+          output_directory_path = File.expand_path(params[:output_directory])
+          message = "Set output directory path to: \"#{output_directory_path}\"."
+          output << message
+          UI.message message
+          ENV['GYM_OUTPUT_DIRECTORY'] = output_directory_path
+          ENV['SCAN_OUTPUT_DIRECTORY'] = output_directory_path
+          ENV['BACKUP_XCARCHIVE_DESTINATION'] = output_directory_path
+        end
+
+        # Set derived data.
+        if params[:derived_data_path]
+          derived_data_path = File.expand_path(params[:derived_data_path])
+          message = "Set derived data path to: \"#{derived_data_path}\"."
+          output << message
+          UI.message message
+          ENV['DERIVED_DATA_PATH'] = derived_data_path # Used by clear_derived_data.
+          ENV['XCODE_DERIVED_DATA_PATH'] = derived_data_path
+          ENV['GYM_DERIVED_DATA_PATH'] = derived_data_path
+          ENV['SCAN_DERIVED_DATA_PATH'] = derived_data_path
+        end
+
+        # Set result bundle.
+        if params[:result_bundle]
+          message = "Set result bundle."
+          output << message
+          UI.message message
+          ENV['GYM_RESULT_BUNDLE'] = "YES"
+          ENV['SCAN_RESULT_BUNDLE'] = "YES"
+        end
+
+        return output
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Setup xcodebuild, gym and scan for easier Jenkins integration"
+      end
+
+      def self.details
+        "- Adds and unlocks keychains from Jenkins 'Keychains and Provisioning Profiles Plugin'.\n" \
+        "- Sets code signing identity from Jenkins 'Keychains and Provisioning Profiles Plugin'.\n" \
+        "- Sets output directory to './output' (gym, scan and backup_xcarchive).\n" \
+        "- Sets derived data path to './derivedData' (xcodebuild, gym, scan and clear_derived_data).\n" \
+        "- Produce result bundle (gym and scan).\n"
+      end
+
+      def self.available_options
+        [
+          # General.
+          FastlaneCore::ConfigItem.new(key: :force,
+                                       env_name: "FL_SETUP_JENKINS_FORCE",
+                                       description: "Force setup, even if not executed by Jenkins",
+                                       is_string: false,
+                                       default_value: false,
+                                       optional: true),
+
+          # Keychain.
+          FastlaneCore::ConfigItem.new(key: :unlock_keychain,
+                                       env_name: "FL_SETUP_JENKINS_UNLOCK_KEYCHAIN",
+                                       description: "Unlocks keychain",
+                                       is_string: false,
+                                       default_value: true,
+                                       optional: true),
```

Comment:
Please remove the `optional: true` if a `default_value` is set for all those options.